src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -289,16 +299,34 @@ Status MutableObjectManager::ReadAcquire(const ObjectID &object_id,
   // Check whether the channel has an error set before checking that we are the only
   // reader. If the channel is already closed, then it's OK to ReadAcquire and
   // ReadRelease in any order.
+  auto now = std::chrono::steady_clock::now();
+  auto timeout_duration = std::chrono::milliseconds(timeout_ms);
+  auto timeout_point = now + timeout_duration;
+  std::chrono::steady_clock::time_point *timeout_point_ptr = nullptr;
+  if (timeout_ms != -1) {
+    RAY_CHECK_GE(timeout_ms, 0);
+    timeout_point_ptr = &timeout_point;
+  }
```

Comment:
Can you decompose this code (along with the same code above) into a separate private method?

Reply:
+1